cmd/byohctl/service/agent.go

```diff
@@ -0,0 +1,645 @@
+	logFile, err := os.OpenFile(logFilePath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, DefaultFilePerms)
+	if err != nil {
+		return utils.LogErrorf("failed to open log file: %v", err)
+	}
+	defer logFile.Close()
+	
+	// Prepare command with restart flag to ensure the agent keeps running even if it doesn't find a machine reference
+	cmd := exec.Command(agentBinary, 
+		"--namespace", namespace,
+		"--metricsbindaddress", "0") // Disable metrics to avoid port conflicts
+	
+	cmd.Stdout = logFile
+	cmd.Stderr = logFile
+	cmd.Dir = byohDir
+	
+	// Set process to a new process group to avoid termination when the parent process exits
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Setpgid: true,
+	}
+	
+	// Start the process
+	utils.LogDebug("Executing command: %s", cmd.String())
+	if err := cmd.Start(); err != nil {
+		return captureAgentDiagnostics(byohDir, agentBinary)
+	}
+	
+	// Wait for a short period to ensure the process doesn't immediately terminate
+	utils.LogInfo("Agent process started with PID: %d", cmd.Process.Pid)
+	utils.LogInfo("Waiting briefly to ensure agent starts properly...")
+	
+	// Don't wait for the process to complete since it should run in the background
+	// Instead, wait for a few seconds to ensure it doesn't immediately crash
+	time.Sleep(2 * time.Second)
+	
+	// Check if the process is still running
+	if cmd.ProcessState != nil && cmd.ProcessState.Exited() {
+		return captureAgentDiagnostics(byohDir, agentBinary)
+	}
+	
+	// Create a systemd service to ensure the agent runs persistently
+	if err := createSystemdService(byohDir, agentBinary, namespace); err != nil {
+		utils.LogWarn("Could not create systemd service: %v - agent may not persist after restart", err)
+	}
+	
+	utils.LogSuccess("BYOH Agent started successfully")
+	return nil
+}
+
+// createSystemdService creates a systemd service file to ensure the agent runs persistently
+func createSystemdService(byohDir, agentBinary, namespace string) error {
+	// Check if we're running as root or have sudo privileges
+	if os.Geteuid() != 0 {
+		utils.LogWarn("Not running as root, skipping systemd service creation")
+		return fmt.Errorf("not running as root")
+	}
+	
+	// Create a wrapper script to ensure proper environment setup
+	wrapperScriptPath := filepath.Join(byohDir, "byoh-agent-wrapper.sh")
+	wrapperContent := fmt.Sprintf(`#!/bin/bash
+# Wrapper script for BYOH Agent to ensure proper environment setup
+export HOME=%s
+cd %s
+exec %s --namespace %s --metricsbindaddress 0
+`, byohDir, byohDir, agentBinary, namespace)
+
+	if err := os.WriteFile(wrapperScriptPath, []byte(wrapperContent), 0755); err != nil {
+		return fmt.Errorf("failed to create wrapper script: %v", err)
+	}
+	
+	// Create systemd service file
+	serviceContent := fmt.Sprintf(`[Unit]
+Description=BYOH Agent Service
+After=network.target
+
+[Service]
+Type=simple
+User=root
+ExecStart=%s
+Restart=always
+RestartSec=10
+WorkingDirectory=%s
+
+[Install]
+WantedBy=multi-user.target
+`, wrapperScriptPath, byohDir)
+
```

Comment:
Why are these files generated in code? Why can't we store them as files directly?

Reply:
Sure!..Segregated these as files